`fast-check` vitest diffing is shallow

## 🐛 Bug Report

From what I can tell, vitest always puts a super shallow diff in the `AssertionError`'s message for a `toEqual` assertion. For example:
```
AssertionError: expected { properties: { '': { …(1) } }, …(2) } to strictly equal { properties: { '': { …(1) } }, …(2) }
```

This isn't a helpful diff because it doesn't show me what's actually different.

## To Reproduce

Use `expect(...).toEqual(...)` between deeply nested objects in a property-based vitest test.

## Expected behavior

I would expect a real diff, like what you normally get with vitest.

From what I can tell, the `AssertionError` object has properties like these:
```
  actual: { properties: { '': [Object] }, type: 'object' },
  expected: { properties: { '': [Object] }, type: 'object' },
  showDiff: true,
  operator: 'deepStrictEqual'
```

I did some research and it appears [`processError` from `@vitest/utils/error` adds a `diff` property to the error](https://github.com/vitest-dev/vitest/blob/bd2c481fe81d9e98575b311d6bc4861f3aca8251/packages/utils/src/error.ts#L165), and maybe that's what we should be including in the thrown message? Not sure why `vitest` isn't picking up the `AssertionError` on its own though (maybe it's wrapped weirdly or something?)

## Your environment

| Packages / Softwares | Version(s) |
| -------------------- | ---------- |
| fast-check           |  3.23.2          |
| @fast-check/vitest | ^0.2.1 |
| node                 |    v23.11.0        |
| TypeScript\*         |   ^5.8.2         |

\*Only for TypeScript's users

--- a/.changeset/smart-needles-brush.md
+++ b/.changeset/smart-needles-brush.md
@@ -1,0 +1,5 @@
+---
+"@fast-check/vitest": patch
+---
+
+🐛(vitest) No shallow diff in Vitest, full diff by default

--- a/packages/vitest/src/internals/TestBuilder.ts
+++ b/packages/vitest/src/internals/TestBuilder.ts
@@ -24,12 +24,15 @@ function adaptParametersForRecord<Ts>(
   parameters: FcParameters<[Ts]>,
   originalParamaters: FcParameters<Ts>,
 ): FcParameters<Ts> {
-  return {
+  const parametersV3OrV4: FcParameters<[Ts]> & { errorWithCause?: boolean } = parameters;
+  const enrichedParameters: FcParameters<Ts> & { errorWithCause?: boolean } = {
     ...(parameters as Required<FcParameters<[Ts]>>),
+    errorWithCause: parametersV3OrV4.errorWithCause !== undefined ? parametersV3OrV4.errorWithCause : true,
     examples: parameters.examples !== undefined ? parameters.examples.map((example) => example[0]) : undefined,
     reporter: originalParamaters.reporter,
     asyncReporter: originalParamaters.asyncReporter,
   };
+  return enrichedParameters;
 }
 
 function adaptExecutionTreeForRecord<Ts>(executionSummary: ExecutionTree<[Ts]>[]): ExecutionTree<Ts>[] {

